File: spring-web-java/src/main/java/spring/web/java/domain/converter/LocalDateTimeConverter.java
package spring.web.java.domain.converter;

import java.time.LocalDateTime;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import spring.web.java.infrastructure.util.DateTimeUtil;

@Converter(autoApply = true)
public class LocalDateTimeConverter implements AttributeConverter<LocalDateTime, LocalDateTime> {

	@Override
	public LocalDateTime convertToDatabaseColumn(LocalDateTime attribute) {
		if (attribute == null) {
			return null;
		}
		return DateTimeUtil.convertKstToUtc(attribute);
	}

	@Override
	public LocalDateTime convertToEntityAttribute(LocalDateTime dbData) {
		if (dbData == null) {
			return null;
		}
		return DateTimeUtil.convertUtcToKst(dbData);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/converter/LocalDateTimeConverter.kt
package spring.web.kotlin.domain.converter

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import spring.web.kotlin.infrastructure.util.convertToKst
import spring.web.kotlin.infrastructure.util.convertToUtc
import java.time.LocalDateTime

@Converter(autoApply = true)
class LocalDateTimeConverter: AttributeConverter<LocalDateTime?, LocalDateTime?> {

    override fun convertToDatabaseColumn(attribute: LocalDateTime?) = attribute?.convertToUtc()

    override fun convertToEntityAttribute(dbData: LocalDateTime?) = dbData?.convertToKst()
}